File: examples/lh-server/go/06-threads/workflow.go
package main

import "github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"

var parentVariable *littlehorse.WfRunVariable

func EntrypointThreadLogic(wf *littlehorse.WorkflowThread) {
	parentVariable = wf.DeclareStr("parent-var").WithDefault("This is the default value of the parent variable")
	wf.Execute("my-task", parentVariable)

	child := wf.SpawnThread(Wflogic, "child", map[string]interface{}{
		"child-input": "This is the input to the child thread",
	})

	wf.Sleep(25)
	wf.Execute("my-task", parentVariable)
	wf.WaitForThreads(child)
}

func Wflogic(wf *littlehorse.WorkflowThread) {
	childInput := wf.DeclareStr("child-input").Required()
	wf.Execute("my-task", childInput)
	wf.Execute("my-task", parentVariable)
	parentVariable.Assign("This is the value of the parent variable set by the child.")
	wf.Sleep(45)
}

File: examples/lh-server/python/06-threads/workflow.py
from littlehorse.workflow import SpawnedThreads, Workflow, WorkflowThread


class MyWorkflow:
    def __init__(self) -> None:
        self.parent_variable = None

    def entrypoint_thread_logic(self, wf: WorkflowThread) -> None:
        self.parent_variable = wf.declare_str("parent-var")
        self.parent_variable.assign("This is the parent variable's initial value")
        wf.execute("my-task", self.parent_variable)
        child = wf.spawn_thread(
            self.child_thread_logic,
            thread_name="child",
            input={"child-input": "This is the input to the child thread"},
        )
        wf.sleep(25)
        wf.execute("my-task", self.parent_variable)
        wf.wait_for_threads(SpawnedThreads.from_list(child))

    def child_thread_logic(self, wf: WorkflowThread) -> None:
        child_input = wf.declare_str("child-input").required()
        wf.execute("my-task", child_input)
        wf.execute("my-task", self.parent_variable)
        self.parent_variable.assign(
            "This is the value of the parent variable set by the child."
        )
        wf.sleep(45)

    def get_workflow(self) -> Workflow:
        return Workflow("threads-example", self.entrypoint_thread_logic)
